R/tests.R

```diff
@@ -352,10 +352,8 @@ testMapping <- function(seed = 12345L, nFeatures = 100,
   model_01_feats <- model_01_feats[order(model_01_feats)]
   model_02_feats <- model_02_feats[order(model_02_feats)]
 
-  set.seed(1)
-  model_01_feats[which(model_01_feats %in% sample(model_01_feats, 10))] <- NA
-  set.seed(2)
-  model_02_feats[which(model_02_feats %in% sample(model_02_feats, 10))] <- NA
+  model_01_feats[c(1, 14, 32, 55, 99, 108)] <- NA
+  model_02_feats[c(6, 18, 30, 75, 88, 102)] <- NA
```

Comment:
Ah, smart. Explicitly setting missing featureIDs avoids any randomness issues.

Could you please update it though so that the elements that are set to `NA` are affected by the argument `nFeatures`? The default is 100, so your code actually increases the length of the vector.

```R
Browse[1]> length(model_01_feats)
[1] 100

model_01_feats[c(1, 14, 32, 55, 99, 108)] <- NA

Browse[2]> length(model_01_feats)
[1] 108

Browse[2]> tail(model_01_feats)
[1] NA NA NA NA NA NA
```

Instead, you can update the missing features like this:

```R
missing_01 <- c(1, 14, 32, 55, 99, 108)
missing_01 <- missing_01[missing_01 <= nFeatures)
# and then the same for missing_02
```

Reply:
Oh, sorry for this miss. Added the proposed, more robust code.